cmd/ig/daemon.go

```diff
@@ -87,11 +116,41 @@ func newDaemonCommand(runtime runtime.Runtime) *cobra.Command {
 
 		log.Infof("starting Inspektor Gadget daemon at %q", socket)
 		service := gadgetservice.NewService(log.StandardLogger(), eventBufferLength)
+
+		var options []grpc.ServerOption
+
+		if serverKey != "" || serverCert != "" || clientCA != "" {
```

Comment:
I think we should we add a more specific validation logic to fail if "one but not all" flags are defined instead of relying on the functions below failing. It'll also print a better error message to the user.